### ISSUE TITLE
Broken Socket Error Handling after PEP 3151 (Python >= 3.3)

### ISSUE THREAD

**Issue by FlorianFranzen** [Tribler/tribler#5069]
Since [PEP 3151](https://www.python.org/dev/peps/pep-3151/) has been implemented with Python 3.3 the error handling for socket-based errors is broken. 
All exceptions thrown by a socket are now of the type ```builtins.OSError``` and ```socket.error``` is now [an alias for the former](https://docs.python.org/3.8/library/socket.html#exceptions). 
Because exceptions are matched based on their message content a change in type also results in a change of message and the exceptions are no longer recognized correctly.

As a fix I would suggest checking ```socket.error.__name__``` to determine the string to use when matching.
 
Please let me know what you think is appropriate and which branch to base it again (noodle seems to have collected some python3-related fixes as well) and I will formulate a matching pull request. 

- [x] I have read CONTRIBUTING.rst

- [ ] I have tried with the latest pre-release version and I still can reproduce the issue.

##### Tribler version/branch+revision: #####
Tribler 7.4.0-exp1 (from nixos-unstable)

##### Operating system and version: #####
NixOS 19.09

##### Steps to reproduce the behavior: #####
Run Tribler under any version of Python >= 3.3

##### Expected behavior: #####
Socket errors are caught inside ```session.py```

##### Actual behavior: #####builtins.OSError
Socket errors go uncaught and crash application

##### Relevant log file output: #####
```
Unhandled error in Deferred:

Traceback (most recent call last):
  File "/nix/store/s7vw8y02cqzx8bnjxl4bkI am still getting this error on NixOS 19.09 andhlnwvz6ws1s-python3-3.7.6/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/nix/store/s7vw8y02cqzx8bnjxl4bkhlnwvzbuiltins.OSError6ws1s-python3-3.7.6/lib/python3.7/threading.py", line 870, in runbuiltins.OSError
    self._target(*self._args, **self._kwargs)
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/_threads/_threadworker.py", line 46, in work
    task()
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/_threads/_team.py", line 190, in doWork
    task()
--- <exception caught here> ---
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.and0/lib/python3.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/nix/store/s1bri1qmkm4nggf4w991si52f02555qn-python3.7-Twisted-19.10.0/lib/python3.7/site-packages/twisted/internet/_resolver.py", line 124, in get
    socketType)
  File "/nix/store/s7vw8y02cqzx8bnjxl4bkhlnwvz6ws1s-python3-3.7.6/lib/python3.7/socket.py", line 752, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
builtins.OSError: [Errno 16] Device or resource busy
```

**Comment by ichorid** [comment]
So, our branch situation is as follows: `release-7.4.0` is the first release based on Python3. However, `devel` is much ahead, especially in that we ditched Twisted in favor of Asyncio. Also, the repository structure changed completely in `devel`, so merges from `release-7.4.0`->`devel` are nearly impossible. This means you'll have to file two PRs: one for `release-7.4.0` and another one for `devel`.

**Comment by FlorianFranzen** [comment]
Well, I guess ```Error 16``` will be caught now correctly, but there are a lot of other exception types and codes in the socket library that will still fall through.

All exceptions of class ```socket.error``` and related are now just aliases for ```builtin.OSError```. This is not fixed yet!

**Comment by ichorid** [comment]
solved by #5120 

at least, to some extent...
